### DISCUSSION TITLE
1020 VMError trying to upgrade a module

### DISCUSSION THREAD

**Discussion by LawsonGraham** [aptos-labs/aptos-developer-discussions#331] [Questions]
### Describe your question in detail.

I am upgrading a module via an msafe txn (don't believe issue is with msafe) and getting a 1020 VMError, and not seeing / understanding what that means. Can someone help?

The changes I've made are modifying an internal inline function's function header, changing an import to not import a type, and modified logic within public functions... I believe all of these are allowed and upgradeable?

### What error, if any, are you getting?

getting a type error, and getting this when trying to upgrade from an msafe multisig...

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

Aptos CLI

**Comment by 0x-j** [comment]
> modifying an internal inline function's function header
Do you mean modifying the parameter or return type of a function? That's considered as breaking change, you can only modify internal logic of any function

**Comment by georgemitenkov** [comment]
Can you please copy-paste your code here (old version vs new version)?
